src/library/dlgtrackinfo.cpp

```diff
@@ -613,6 +613,13 @@ void DlgTrackInfo::saveTrack() {
     slotSpinBpmValueChanged(spinBpm->value());
     static_cast<void>(updateKeyText()); // discard result
 
+    // Don't update if nothing changed. This prevents wiping the record
+    // when calling this function twice quickly.
+    // See https://github.com/mixxxdj/mixxx/issues/12963
+    if (m_trackRecord == m_pLoadedTrack->getRecord()) {
+        return;
+    }
+
     // Update the cached track
     // The dialog is updated and repopulated by the Track::changed() signal.
     m_pLoadedTrack->replaceRecord(std::move(m_trackRecord), std::move(m_pBeatsClone));
```

Comment:
Sorry, my bad.

```suggestion
    // Update the cached track
    //
    // If replaceRecord() returns true then both m_trackRecord and m_pBeatsClone
    // will be updated by the subsequent Track::changed() signal to keep them
    // synchronized with the track. Otherwise the track has not been modified and
    // both members must remain valid. Do not use std::move() for passing arguments!
    m_pLoadedTrack->replaceRecord(m_trackRecord, m_pBeatsClone);
```

Without warranty. Consider refactoring the API to prevent such misunderstandings in the future.

Reply:
Thanks for jumping in!

Okay, so this is the actual fix.
IIUC the no-op test I added above just hides the issue because it's unlikely the record (in DlgTrackInfo) has changed in between two quick save() calls. Correct?

Reply:
Also, IIUC this
https://github.com/mixxxdj/mixxx/blob/e2d41999e7c017c6a353f87967152d3bd2102354/src/track/track.cpp#L308-L310 would already `move` the beats even if applying them fails and replaceRecord() returns true?

Reply:
It only moves out of the argument which is passed by value. No side effects on the outer context.